Made payment page, brandCode and skipSelection overwritable

onchainguy-btc opened the pull request.

Commit:
Made payment page, brandCode and skipSelection overwritable

Commit:
Removed superfluous space in payment page filename

hekike commented:
@seb0zz looks good but one of the test is broken. I'm not using Adyen a long time ago. Could you help to fix the test as well?

onchainguy-btc commented:
Yeah, I will take a look at the tests within the following days!

hekike commented:
@seb0zz thank you!

onchainguy-btc commented:
Mhhh might it be that the merchant account does not exist anymore or has been changed? The test fails due to the following error:

Error: Merchant signature was not valid!<br />You can test the HMAC Calculation by submitting a payment request to https://ca-test.adyen.com/ca/ca/skin/checkhmac.shtml

hekike commented:
@seb0zz it's possible. I don't remember where I found that test account. Maybe we should mock it out. What do you think?

onchainguy-btc commented:
Well in fact I think we don't even need to test the response from adyen but only that the `generateRequest` method is generating a valid URL for adyen. So instead of checking the HTML body, we could just check if the URL contains the `merchantReference` etc. What do you think?

hekike commented:
@seb0zz yes but the idea was behind that checking the validity of the generated request. But let's to this, better then no running test yet :D

onchainguy-btc commented:
Sorry for the delay, I think I can do it tomorrow!

Commit:
Some minor test refactorings

As the adyen account that was used for testing does not exist anymore, we cannot use a real payment window for testing. In fact the window should be mocked out anyways. For now it should be fine to verify that the URL for adyen is properly build up.

onchainguy-btc commented:
Alright, what do you think? Merge? :-)

hekike commented on test/hpp.js:
```diff
@@ -37,23 +36,18 @@ describe('HPP', function () {
 
     it('should generate url and merchantReference', function (done) {
       hppPayment.generateRequest(function (err, url, merchantReference) {
+        console.log(url);
```
please remove console.log

hekike commented:
looks good, only a small comment ;)

Commit:
Removed console.log

onchainguy-btc commented:
Oh no I'm sorry :eyes: removed the log output!

hekike commented:
v1.0.0 rolled out. Yes it wasn't proper SemVer :P my fault...
https://www.npmjs.com/package/adyen ;)

hekike commented:
Thanks for the collaboration!

onchainguy-btc commented:
Thanks man! :heart_eyes:

hekike commented:
Oh one more thing, could you put to the README in an other PR as well? I'm pretty sure that would be helpful for the others.
